src/Build/ObjectModelRemoting/DefinitionObjectsLinks/ProjectLink.cs

```diff
@@ -259,5 +259,11 @@ public abstract class ProjectLink
         /// Called by the local project collection to indicate to this project that it is no longer loaded.
         /// </summary>
         public abstract void Unload();
+
+        public virtual bool GlobalPropertiesContains(string key) => GlobalProperties.ContainsKey(key);
```

Comment:
Comment that these are default implementations to avoid breaks but an implementing class may wish to override for perf?

Reply:
Good point, and I should do that for our overrides, too.